tcp/segment.ml

```diff
@@ -97,9 +97,9 @@ module Rx(Time:V1_LWT.TIME) = struct
       with Not_found -> false *)
 
   (* Determine the transmit window, from the last segment *)
-  let window q =
+  let window ~default q =
     try (S.max_elt q).window
-    with Not_found -> 0
+    with Not_found -> default
```

Comment:
This looks correct to me. From an API perspective, it might be slightly nicer to explicitly differentiate between the default window and the actual window.  If this function returned an `option` instead, then this would be easy to do in the caller.